### DISCUSSION TITLE
Parsec service 1.3.0 release

### DISCUSSION THREAD

**Discussion by gowthamsk-arm** [parallaxsecond/parsec#704] [General]
As we approach the next release of Parsec following [an established process](https://parallaxsecond.github.io/parsec-book/contributing/release_process.html), Parsec will be released with v1.3.0, and we are aiming for Mid October.

Use this thread for discussions about the 1.3.0 release of the Parsec service, including:

- the timeline of the release
- the status of the work feeding into the release

Date of the Code Freeze (Tentative): 29th of September

Stages:

- [x] Prior Code freeze
  - [x] Notify the community about the code-freeze date, release date, and the release discussion on GitHub
  - [x] Review PRs across Parsec repos 
  - [x] Merge approved PRs 
  - [x] Publish new versions of all managed crates and configure parsec service and tool to use them. See this [guide](https://parallaxsecond.github.io/parsec-book/contributing/package_management.html).
  - [x] Run cargo update for parsec service and tool
  - [x] Manual review of the crates versions in parsec/cargo.toml and parsec-tool/cargo.toml 
  - [x] Check common dependency crates versions between the parsec service, parsec tool, and other crates
  - [x] Make sure there are no critical threats in the system 
  - [x] Check the milestone list
  
- [x] During code freeze
  - [x] Issue release candidate tag 1.3.0-rc1 
  - [x] OpenSUSE packaging for the RC
  - [x] Pull in the latest git dependencies in the crates to ensure the latest release would work.
  - [x] Execute the normal e2e tests on hardware:
    - [x] Raspberry Pi with mbed-crypto 
    - [x] Raspberry Pi with TPM
    - [x] Raspberry Pi with Nitrokey HSM (pluggable USB)
    - [x] N1SDP with Trusted Services
  - [x] Execute fuzz testing for some time
  - [x] Ask the Linux distributions packagers to try the tagged version
  - [x] Ask the Parsec Yocto maintainers to try the tagged version
  - [x] Ask any interested people to try the tagged version
  - [x] Review the book and make sure all pages are up-to-date. In particular, make sure the Threat Model is up to date.
  - [x] Make sure all the markdown pages (like the README) look good.
  - [x] Make sure the rustdoc pages (on docs.rs) are looking good (before and after the release).
  - [x] Update parsec book to change references from 1.2.0 to 1.3.0
  - [x] If any bugs are reported, we fix bugs and then issue another release candidate,1.3.0-rcx
  - [x] When All reported issues are addressed, make a GitHub release out of the last tag, which includes its changelog, and ask maintainers to release this version of Parsec.
  
- [x] Release
  - [x] Push the release tag 1.3.0 or the service and 0.7.0 for the parsec-tool
  - [x] Publish the new quickstart bundle and update the quickstart guide with it
 
 - [ ] Post-release
    - [ ] Close issues fixed in the release and the past milestone
    - [x] Update, test, and upstream Yocto recipes for Parsec and parsec-tool
    - [ ] Make a new issue milestone for the next release
    - [ ] Update the [Roadmap](https://github.com/parallaxsecond/community/blob/main/ROADMAP.md)
    - [ ] Update the [Parsec Release Process](https://parallaxsecond.github.io/parsec-book/contributing/release_process.html) with what we learned.
    - [x] If any bugs are reported, make new MINOR or PATCH tags (that would be {releave-version}.x  for this release)
    - [x] Update the Parsec Tool demo
    - [ ]  Profit :clinking_glasses:

**Comment by tgonzalezorlandoarm** [comment]
Release candidates for PARSEC:

- https://github.com/parallaxsecond/parsec/tree/1.3.0-rc1
- https://github.com/parallaxsecond/parsec/tree/1.3.0-rc2

Release candidates for PARSEC-TOOL:

- https://github.com/parallaxsecond/parsec-tool/tree/0.7.0-rc1

**Comment by gowthamsk-arm** [comment]
Parsec 1.3 recipes in meta-security `master-next`. 
https://gitlab.com/akuster/meta-security/-/commit/77be5c4219c6733bea0ab7f2667b53f1c3371a80
